internal/ignition/ignition.go

```diff
@@ -0,0 +1,378 @@
+func (g *installerGenerator) modifyBMHFile(file *config_31_types.File, bmh *bmh_v1alpha1.BareMetalHost, host *models.Host) error {
+	inventory := models.Inventory{}
+	err := json.Unmarshal([]byte(host.Inventory), &inventory)
+	if err != nil {
+		return err
+	}
+
+	hw := bmh_v1alpha1.HardwareDetails{
+		CPU: bmh_v1alpha1.CPU{
+			Arch:           inventory.CPU.Architecture,
+			Model:          inventory.CPU.ModelName,
+			ClockMegahertz: bmh_v1alpha1.ClockSpeed(inventory.CPU.Frequency),
+			Flags:          inventory.CPU.Flags,
+			Count:          int(inventory.CPU.Count),
+		},
+		Hostname: host.RequestedHostname,
+		NIC:      make([]bmh_v1alpha1.NIC, len(inventory.Interfaces)),
+		Storage:  make([]bmh_v1alpha1.Storage, len(inventory.Disks)),
+	}
+	if inventory.Memory != nil {
+		hw.RAMMebibytes = int(inventory.Memory.PhysicalBytes / 1024 / 1024)
+	}
+	for i, iface := range inventory.Interfaces {
+		hw.NIC[i] = bmh_v1alpha1.NIC{
+			Name:      iface.Name,
+			Model:     iface.Product,
+			MAC:       iface.MacAddress,
+			SpeedGbps: int(iface.SpeedMbps / 1024),
+		}
+		switch {
+		case len(iface.IPV4Addresses) > 0:
+			hw.NIC[i].IP = iface.IPV4Addresses[0]
+		case len(iface.IPV6Addresses) > 0:
+			hw.NIC[i].IP = iface.IPV6Addresses[0]
+		}
+	}
+	for i, disk := range inventory.Disks {
+		hw.Storage[i] = bmh_v1alpha1.Storage{
+			Name:         disk.Name,
+			Vendor:       disk.Vendor,
+			SizeBytes:    bmh_v1alpha1.Capacity(disk.SizeBytes),
+			Model:        disk.Model,
+			WWN:          disk.Wwn,
+			HCTL:         disk.Hctl,
+			SerialNumber: disk.Serial,
+			Rotational:   (disk.DriveType == "HDD"),
+		}
+	}
+	if inventory.SystemVendor != nil {
+		hw.SystemVendor = bmh_v1alpha1.HardwareSystemVendor{
+			Manufacturer: inventory.SystemVendor.Manufacturer,
+			ProductName:  inventory.SystemVendor.ProductName,
+			SerialNumber: inventory.SystemVendor.SerialNumber,
+		}
+	}
+	status := bmh_v1alpha1.BareMetalHostStatus{
+		HardwareDetails: &hw,
+		PoweredOn:       true,
+	}
+	statusJSON, err := json.Marshal(status)
+	if err != nil {
+		return err
+	}
+	metav1.SetMetaDataAnnotation(&bmh.ObjectMeta, bmh_v1alpha1.StatusAnnotation, string(statusJSON))
+
+	serializer := k8sjson.NewSerializerWithOptions(
+		k8sjson.DefaultMetaFactory, nil, nil,
+		k8sjson.SerializerOptions{
+			Yaml:   true,
+			Pretty: true,
+			Strict: true,
+		},
+	)
+	buf := bytes.Buffer{}
+	err = serializer.Encode(bmh, &buf)
+	if err != nil {
+		return err
+	}
+
+	encodedBMH := base64.StdEncoding.EncodeToString(buf.Bytes())
+	source := "data:text/plain;charset=utf-8;base64," + encodedBMH
+	file.Contents.Source = &source
+
+	return nil
+}
+
+// sortHosts sorts hosts into masters and workers, excluding disabled hosts
+func sortHosts(hosts []*models.Host) ([]*models.Host, []*models.Host) {
+	masters := []*models.Host{}
+	workers := []*models.Host{}
+	for i := range hosts {
+		switch {
+		case hosts[i].Status != nil && *hosts[i].Status == models.HostStatusDisabled:
```

Comment:
I think @yevgeny-shnaidman  just changed this logic to:
Exclude the disabled hosts.
First sorting key is host role (Master < Worker), secondary sorting key is Hostname.

Reply:
What is the purpose of sorting the individual hosts by hostname?

Reply:
I added sorting and a comment explaining it, which assumes it's for the sake of repeatable results. I looked at the python code and PR that introduced the change, and it's not obvious why the sort is happening there.